woocommerce/woocommerce
# CYS - add shuffle feature

> [!WARNING]  
> This PR is blocked by #47322


### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR allows the user to shuffle patterns of the same category. Currently, the only category available is WooCommerce, and for this reason, header/footer patterns will be rendered.

We will introduce multiple categories with https://github.com/woocommerce/woocommerce/issues/47328.


https://github.com/woocommerce/woocommerce/assets/4463174/5d037ddc-6e33-4e90-b063-33afbd1a43cc




Closes #47325 and #47326.


**Note**: Shuffling option is not displayed on the first template because the pattern has to be parsed before. Currently, we parsed them on the homepage preview: https://github.com/woocommerce/woocommerce/blob/cca31dfc1862d6f10495b97fc3de8a798b49510e/plugins/woocommerce-admin/client/customize-store/assembler-hub/sidebar/sidebar-navigation-screen-homepage.tsx/#L120-L125

We should revisit/add this logic when we will migrate patterns to PTK #45835 





### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:


1. Ensure that you have Gutenberg enabled.
2. Make sure you have the WooCommerce beta tester plugin installed
3. Head over to Tools > WCA Test Helper > Features
4. Make sure you see the `pattern-toolkit-full-composability` on the list.
5. Enable it.
6. Head over to WooCommerce > Home > Customize your store.
7. Click on "Start designing" and proceed to the Pattern Assembler.
8. Click on "Design your Homepage".
9. Switch Pattern.
10. Ensure that the block toolbar appears, and you can shuffle patterns.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

CYS: add shuffle feature.

#### Comment 



</details>
